### DISCUSSION TITLE
features

### DISCUSSION THREAD

**Discussion by elias3433** [NickvisionApps/Tagger#420] [Ideas]
Hello,
I recently discovered your software which I really like.
I suggest some improvements and additional functions:
- be able to tag songs automatically like with audioranger under Windows that one of my friends uses.
for my part I don't like windows and use manjaro.
Can you use Shazam or Songrec directly in tagger?
- be able to add a cover by dragging and dropping a file or retrieving it directly from the internet
- be able to rename file names like Kid3 does from metadata

I find it tedious to have to use several software programs for common operations

In any case, congratulations for your software.

Best  regards

elias

**Comment by nlogozzo** [comment]
Hi, all of these are features already.

> be able to tag songs automatically like with audioranger under Windows that one of my friends uses.

This is the Download Metadata from MusicBrainz feature. This can be accessed on Windows by going to the Tag Menu -> Web Services 

> retrieving it directly from the internet

The Download Metadata from MusicBrainz will pull cover art if it's availability 

> be able to rename file names like Kid3 does from metadata

This is the Tag to File Name feature that you can find on Linux via the tag action menu (the icon with the wrench in the bottom bar) -> Convert. On Windows Tag Menu -> Convert
